de.peeeq.wurstscript/src/main/java/de/peeeq/wurstio/jassinterpreter/providers/GroupProvider.java

```diff
@@ -74,4 +74,35 @@ public ILconstInt BlzGroupGetSize(IlConstHandle group) {
         LinkedHashSet<IlConstHandle> groupList = (LinkedHashSet<IlConstHandle>) group.getObj();
         return ILconstInt.create(groupList.size());
     }
+	
+    public ILconst BlzGroupUnitAt(IlConstHandle group, ILconstInt index) {
+        LinkedHashSet<IlConstHandle> groupList = (LinkedHashSet<IlConstHandle>) group.getObj();
+        ILconst elem = ILconstNull.instance();
+        if(index.getVal() > 0 && index.getVal() < groupList.size()) {
+            for(int i = 0; i < index.getVal(); i++)
+                elem = groupList.iterator().next();
```

Comment:
This looks like it would always  return the first element of the group, since you call `iterator` inside the loop. Maybe you can do another pull request to fix this?